### DISCUSSION TITLE
Question about defining a problem

### DISCUSSION THREAD

**Discussion by hkim716** [pyg-team/pytorch_geometric#7086] [Q&A] [answered]
I'd like to predict delay time between several cities using GNN. Several cities can be linked each other so we can have large directional graph.
In this case, one transportation can be thought of 1 directioanal edge, and each station will have several node features like departure time schedule, and each edge will have edge attributes like delay time, average speed, and so on.
Furthermore, since there will be more than one transportation for the same route, there should be more than one edges between the same two nodes. 
In this case, is it considered to be link prediction problem? How could I simplify the problem for using GNN?

**Comment by rusty1s** [comment, marked as answer]
I think this would be more of an edge regression problem, since I assume you do not necessarily predict the existence of a link, but more of a regression value between two cities. I think this should be well implementable within the PyG framework, where you can use a GNN layer that can incorporate multi-dimensional edge features (e.g., `SplineConv, `TransformerConv`, etc), and then obtain a node embedding for source and destination city to make a final prediction for the link between the two (using some kind of MLP on top of concatenated source and destination node embeddings).
